src/net/mod.rs

```diff
@@ -10,6 +10,7 @@
 mod send_recv;
 mod socket;
 mod socket_addr_any;
+mod socket_address;
```

Comment:
Could we make this a `pub mod`, and perhaps name it `addr`, to keep `SockAddr` and `SockAddrRaw` from cluttering the rest of the `net` API? It seems like most users won't ever need to use those directly.

Reply:
Is it confusing that `rustix::net::addr` then only contains those two, and not any of the other address types (`SocketAddrUnix`, `SocketAddrV4`, `SocketAddrV6`, `SocketAddr`, `SocketAddrAny`, `SocketAddrStorage`) that are one level up? Seems like at least `SocketAddrStorage` and `SocketAddrOpaque` should be next to each other since they represent similar C types.

Maybe move them all into `rustix::net::addr`?

Reply:
Good idea to keep `SocketAddrStorage` next to `SocketAddrOpaque`.

It seems nice to keep `SocketAddrV4`, `SocketAddrV6`, `SocketAddrUnix`, in `rustix::net` where all the rest of the API that uses them are though, because that's the "regular" API.

What if we put `SocketAddrOpaque`, `SocketAddrArg`, and `SocketAddrStorage` into a `pub mod` named `rustix::net::abstract`?

Reply:
Sounds good. `abstract` might be confused with the abstract namespace for unix sockets, but I don't have a better name to propose.

Reply:
Also, `abstract` is a Rust reserved keyword.